File: scripts/render-icons.py
import os
from svglib.svglib import svg2rlg
from reportlab.graphics import renderPM
from reportlab.lib.colors import Color

# Paths
base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
svg_path = os.path.join(base_dir, 'public', 'icons', 'notification-icon.svg')
icon_path = os.path.join(base_dir, 'public', 'icons', 'notification-icon.png')
badge_path = os.path.join(base_dir, 'public', 'icons', 'notification-badge.png')

# Transparency background
transparent = Color(0, 0, 0, 0)

def render_svg(out_path, size):
    # Load fresh drawing to avoid cumulative scaling
    drawing = svg2rlg(svg_path)
    
    # Scale drawing
    scale_factor = size / 512.0
    drawing.width = size
    drawing.height = size
    drawing.scale(scale_factor, scale_factor)
    
    # Render drawing with transparent background
    renderPM.drawToFile(drawing, out_path, fmt='PNG', bg=transparent)

print("Rendering 192x192 icon...")
render_svg(icon_path, 192)
print("Rendering 72x72 badge...")
render_svg(badge_path, 72)
print("Completed successfully!")

File: scripts/render-icons.js
const { chromium } = require('@playwright/test');
const path = require('path');

async function main() {
    const browser = await chromium.launch({
        executablePath: 'C:\\Users\\Arafat Mbaga\\AppData\\Local\\ms-playwright\\chromium_headless_shell-1228\\chrome-headless-shell-win64\\chrome-headless-shell.exe'
    });
    const page = await browser.newPage();

    const svgPath = path.join(__dirname, '../public/icons/notification-icon.svg');
    const absoluteSvgUrl = 'file:///' + svgPath.replace(/\\/g, '/');
    
    console.log('Loading SVG from:', absoluteSvgUrl);
    await page.goto(absoluteSvgUrl);

    // Render 192x192 icon
    await page.setViewportSize({ width: 192, height: 192 });
    const iconPath = path.join(__dirname, '../public/icons/notification-icon.png');
    const svgEl = page.locator('svg');
    await svgEl.screenshot({ path: iconPath, omitBackground: true });
    console.log('Saved 192x192 icon to:', iconPath);

    // Render 72x72 badge
    await page.setViewportSize({ width: 72, height: 72 });
    const badgePath = path.join(__dirname, '../public/icons/notification-badge.png');
    await svgEl.screenshot({ path: badgePath, omitBackground: true });
    console.log('Saved 72x72 badge to:', badgePath);

    await browser.close();
}

main().catch(err => {
    console.error('Error rendering icons:', err);
    process.exit(1);
});
